src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -3023,14 +3023,17 @@ def import_ssl_cert(cmd, resource_group_name, name, key_vault, key_vault_certifi
     kv_secret_name = None
     cloud_type = cmd.cli_ctx.cloud.name
     from azure.cli.core.commands.client_factory import get_subscription_id
+    from azure.cli.core.profiles import AD_HOC_API_VERSIONS
     subscription_id = get_subscription_id(cmd.cli_ctx)
     if cloud_type.lower() == PUBLIC_CLOUD.lower():
         if kv_subscription.lower() != subscription_id.lower():
             diff_subscription_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_APPSERVICE,
                                                                subscription_id=kv_subscription)
-            ascs = diff_subscription_client.app_service_certificate_orders.list()
+            ascs = diff_subscription_client.app_service_certificate_orders.list(
+                api_version=AD_HOC_API_VERSIONS[ResourceType.MGMT_APPSERVICE]['app_service_certificate_orders'])
```

Comment:
`AD_HOC_API_VERSIONS` was introduced to preserve certain API versions while trimming mgmt SDKs (#17816, #23946).

As a side note, though your current code is better than hard-coding API version `2022-09-01` here, there is no hard requirement that you must use an ad-hoc API version as `AD_HOC_API_VERSIONS[ResourceType.MGMT_APPSERVICE]['app_service_certificate_orders']`.

For example, `acr` just hard-codes them:

https://github.com/Azure/azure-cli/blob/09a6fff1a34447dbf0987621ca89ab1c1911529d/src/azure-cli/azure/cli/command_modules/acr/_client_factory.py#L8-L13

#25418 also hard-codes `2022-01-31-preview`:

https://github.com/Azure/azure-cli/blob/cbb33a92a291e86e877ff199a738fe8ec17acda2/src/azure-cli/azure/cli/command_modules/identity/_client_factory.py#L18

It is acceptable to use `2022-09-01` here, as long as the hard-coded API version is the same as what is defined in `AD_HOC_API_VERSIONS`.

So, no action is needed! 🙂